File: app/src/main/java/com/mmc/lamandys/liba_datapick/AutoApplication.java
package com.mmc.lamandys.liba_datapick;

import android.annotation.SuppressLint;
import android.app.Application;
import android.content.Context;

import com.mmc.lamandys.liba_datapick.helper.AutoTrackHelper;

import io.flutter.embedding.engine.FlutterEngine;
import io.flutter.embedding.engine.FlutterEngineCache;
import io.flutter.embedding.engine.dart.DartExecutor;

import static com.mmc.lamandys.liba_datapick.Constanats.CACHED_FLUTTER_ENGINE;

@SuppressLint("StaticFieldLeak")
public class AutoApplication extends Application {
    private static Context mContext;
    private static AutoApplication mApp;
    public FlutterEngine mFlutterEngine;

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
    }

    @Override
    public void onCreate() {
        super.onCreate();
        mContext = getApplicationContext();
        mApp = this;
//        AutoTrackHelper.frameDetection();
        AutoTrackHelper.frameDetection2();
        CrashHandler.getinstance().init();

        // Somewhere in your app, before your FlutterFragment is needed, like in the
        // Application class ...
        // Instantiate a FlutterEngine.
        mFlutterEngine = new FlutterEngine(mContext);
        // Start executing Dart code in the FlutterEngine.
        mFlutterEngine.getDartExecutor().executeDartEntrypoint(
                DartExecutor.DartEntrypoint.createDefault()
        );

        // Cache the pre-warmed FlutterEngine to be used later by FlutterFragment.
        FlutterEngineCache.getInstance().put(CACHED_FLUTTER_ENGINE, mFlutterEngine);
    }

    public static AutoApplication getInstance() {
        return mApp;
    }

    public static Context getApplicationContex() {
        return mContext;
    }

}

File: app/src/main/java/com/mmc/lamandys/liba_datapick/AutoApplication.kt
package com.mmc.lamandys.liba_datapick

import android.annotation.SuppressLint
import android.app.Application
import android.content.Context
import android.widget.Toast
import com.mmc.lamandys.liba_datapick.helper.AutoTrackHelper
import io.flutter.embedding.engine.FlutterEngine
import io.flutter.embedding.engine.FlutterEngineCache
import io.flutter.embedding.engine.dart.DartExecutor
import io.flutter.plugin.common.MethodChannel

const val ENGINE_ID = "1"

@SuppressLint("StaticFieldLeak")
class AutoApplication : Application() {

    var count = 0

    private lateinit var channel: MethodChannel

    override fun onCreate() {
        super.onCreate()

        //AutoTrackHelper.frameDetection();
        AutoTrackHelper.frameDetection2()
        CrashHandler.getinstance().init()

        val flutterEngine = FlutterEngine(this)
        flutterEngine
                .dartExecutor
                .executeDartEntrypoint(
                        DartExecutor.DartEntrypoint.createDefault()
                )

        FlutterEngineCache.getInstance().put(ENGINE_ID, flutterEngine)

        channel = MethodChannel(flutterEngine.dartExecutor, Constants.Method_channel)

        channel.setMethodCallHandler { call, _ ->
            when (call.method) {
                Constants.method_increment_counter -> {
                    count++
                    Toast.makeText(this, "count = $count", Toast.LENGTH_SHORT).show()
                    reportCounter()
                }
                Constants.method_request_counter -> {
                    reportCounter()
                }

                Constants.method_finish -> {
                    Toast.makeText(this, "我跳转了", Toast.LENGTH_SHORT).show()
                }
            }
        }
    }

    private fun reportCounter() {
        channel.invokeMethod("reportCounter", count)
    }

    companion object {
        var mApplicationContext: Context? = null
        var instance: AutoApplication? = null
    }
}